metadb/client.go

```diff
@@ -0,0 +1,11 @@
+package metadb
+
+import "context"
+
+// Client exposes the interfaces for metadb service
+type Client interface {
+	// Query execute a query, not support multple statements
```

Comment:
I think this metaDb is quite generic, not specified for our case.
For our case, I don't think we need a complex capacity of full SQL.
Simply Write(key, value)/Read(key, value) is fine for us.
and value will contains every metadata associate with key.

And for more generic, we can have like Write(ctx, namespace, key, value)/ Read(ctx, namespace, key, value).
Namespace can be like type, table,..
How do you think?

Reply:
or more specific, metadb can provide service specific to our system like:
AddThumbnail(id, &Thumbnail{small, medium,large})
GetThumbnail(id) -> *Thumbnail
AddMetadata(id)..
GetMetadata(id)...

And details of implementation can be easy to map to any kind of database(rqlite in our case), like create Thumbnail table should be inside code of metadb than afterFun is set up in AfterFunc()